### PR TITLE
AWS AL2023 support

### PR DESCRIPTION
**Pull request by viktor-kurchenko**
This PR aims to introduce transparent support for installing Cilium on clusters using the AL2023 AWS node image family.

Please see per commit for full details :)

### CODE REVIEW

**Review by gandro — APPROVED** [review]

**Review by brlbil — APPROVED** [review]

**Review by derailed — APPROVED** [review]

File: Documentation/installation/k8s-install-helm.rst [hunk 1]
```diff
@@ -163,8 +163,10 @@ Install Cilium
                      && iptables -t nat -F AWS-CONNMARK-CHAIN-1
 
          Some Linux distributions use a different interface naming convention.
-         If you use masquerading with the option ``egressMasqueradeInterfaces=eth0``,
-         remember to replace ``eth0`` with the proper interface name.
+         If you use masquerading with the option ``egressMasqueradeInterfaces=eth+``,
+         remember to replace ``eth+`` with the proper interface name. For
+         reference, Amazon Linux 2 uses ``eth+``, whereas Amazon Linux 2023 uses
+         ``ens+``.
```
Comment (auriaave) [inline Documentation/installation/k8s-install-helm.rst]: I think the docs should state that mixed node image families with different interface naming conventions is not currently supported.

Comment (viktor-kurchenko) [reply Documentation/installation/k8s-install-helm.rst]: I've added a [quote](https://github.com/cilium/cilium/pull/36076/files#diff-13f16293a6e865c96800d64660bcd4bee961e7b8a60f4b2d4790249bdfe5152eR169).
Thanks!

**Review by joestringer — APPROVED** [review]
Docs changes LGTM.

Were you able to manually confirm that the `.github/actions` changes for setting up EKS clusters will pass CI? I'm not sure whether the CI will pick those up from this PR by default.

**Comment by viktor-kurchenko** [comment]
> Were you able to manually confirm that the `.github/actions` changes for setting up EKS clusters will pass CI? I'm not sure whether the CI will pick those up from this PR by default.

Yeah, CI picks the action from the branch (because it's not an upstream).
Here we go: https://github.com/cilium/cilium/actions/runs/11975475141/job/33388949078#step:20:37
```
✅ Detected AWS node image family: AmazonLinux2023
```

BTW, @joestringer it will be interesting to get your feedback on [this fix](https://github.com/cilium/cilium/pull/36103) as well.

**Comment by viktor-kurchenko** [comment]
Kind ping: @auriaave @nbusseneau @learnitall  ))

**Comment by auriaave** [comment]
I can't approve it (community user) but +1 from me 🙏

**Review by auriaave — APPROVED** [review]

**Review by learnitall — APPROVED** [review]
LGTM! My apologies for the delay

**Comment by brlbil** [comment]
Should this be backported?

**Comment by viktor-kurchenko** [comment]
> Should this be backported?

Not sure, I've asked our SA.

**Comment by joestringer** [comment]
We don't maintain multiple branches for cilium-cli, so there should be no need to backport.

**Comment by gandro** [comment]
We might want to backport just the docs changes though

**Comment by nbusseneau** [comment]
> We don't maintain multiple branches for cilium-cli, so there should be no need to backport.

But `cilium-cli` is part of the tree now, so we should backport for 1.16 IMO.

**Comment by joestringer** [comment]
> We might want to backport just the docs changes though

Yeah strictly speaking we probably should have had a separate PR for the docs to simplify review & backporting. Backporting the docs seems reasonable enough. Given that it's all in one PR though I'd rather the author take care of this. Simply marking it for backport could cause some hassle for backporters since it's all wrapped together.

> But cilium-cli is part of the tree now, so we should backport for 1.16 IMO 🤔It wasn't in 1.16, oops.

I'm realising now that although the commits are all related to AWS AL2023, there's really three different logical changes here and we might make different backporting decisions for each - GitHub actions changes for how we perform testing, docs changes for how we instruct users to properly configure Cilium, and the CLI changes for better AWS platform detection.

**Pull request merged**